pyintegration/deephaven2/dtypes/_datetime.py

```diff
@@ -0,0 +1,108 @@
+#
+#   Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+from typing import Any, Sequence, Callable
+
+from deephaven2 import DHError
+from deephaven2.dtypes import DType
+
+_DHDateTime = DType(j_name="io.deephaven.time.DateTime")
+_DHPeriod = DType(j_name="io.deephaven.time.Period")
+
+
+class DateTime:
+    is_primitive = False
+    j_type = _DHDateTime.j_type
+    qst_type = _DHDateTime.qst_type
+
+    def __init__(self, v: Any):
+        if isinstance(v, _DHDateTime.j_type):
+            self.j_datetime = v
+        else:
+            # TODO conversion support from Python built-in datetime, string datetime representation etc.
+            self.j_datetime = _DHDateTime(v)
+
+    def __eq__(self, other):
+        if other is None:
+            if self.j_datetime is None:
+                return True
```

Comment:
This is confusing to me - we want `DateTime(None) == None`?

Reply:
Can / should `j_datetime` even be allowed to be `None`?

Larger question in general - should we allow wrapped `None` for any types? It means we have to handle:

`None`, `Wrapped(None)`, and `Wrapped(<real>)`  everywhere instead of
`None` and `Wrapped(<real>)`

Reply:
Good catch. I actually went back and forth on this a bit and decided against allowing j_datetime to be None. A null Java object is always translated to None in Python by JPY. I modified the constructor but overlooked __eq__.